src/backend/stream.rs

```diff
@@ -1,47 +1,44 @@
-// Copyright © 2017 Mozilla Foundation
+// Copyright © 2017-2018 Mozilla Foundation
 //
 // This program is made available under an ISC-style license.  See the
 // accompanying file LICENSE for details.
 
 use backend::*;
 use backend::cork_state::CorkState;
-use cubeb;
+use cubeb_backend::{ffi, log_enabled, ChannelLayout, DeviceId, DeviceRef, Error, ForeignTypeRef, Result, SampleFormat,
+                    StreamOps, StreamParamsRef, StreamPrefs};
 use pulse::{self, CVolumeExt, ChannelMapExt, SampleSpecExt, StreamLatency, USecExt};
 use pulse_ffi::*;
+use std::{mem, ptr};
 use std::ffi::{CStr, CString};
 use std::os::raw::{c_long, c_void};
-use std::ptr;
+
 
 const PULSE_NO_GAIN: f32 = -1.0;
 
-fn cubeb_channel_to_pa_channel(channel: cubeb::Channel) -> pa_channel_position_t {
-    assert_ne!(channel, cubeb::CHANNEL_INVALID);
-
-    // This variable may be used for multiple times, so we should avoid to
-    // allocate it in stack, or it will be created and removed repeatedly.
-    // Use static to allocate this local variable in data space instead of stack.
-    static MAP: [pa_channel_position_t; 10] = [
-        // PA_CHANNEL_POSITION_INVALID,      // CHANNEL_INVALID
-        PA_CHANNEL_POSITION_MONO,         // CHANNEL_MONO
-        PA_CHANNEL_POSITION_FRONT_LEFT,   // CHANNEL_LEFT
-        PA_CHANNEL_POSITION_FRONT_RIGHT,  // CHANNEL_RIGHT
-        PA_CHANNEL_POSITION_FRONT_CENTER, // CHANNEL_CENTER
-        PA_CHANNEL_POSITION_SIDE_LEFT,    // CHANNEL_LS
-        PA_CHANNEL_POSITION_SIDE_RIGHT,   // CHANNEL_RS
-        PA_CHANNEL_POSITION_REAR_LEFT,    // CHANNEL_RLS
-        PA_CHANNEL_POSITION_REAR_CENTER,  // CHANNEL_RCENTER
-        PA_CHANNEL_POSITION_REAR_RIGHT,   // CHANNEL_RRS
-        PA_CHANNEL_POSITION_LFE           // CHANNEL_LFE
-    ];
-
-    let idx: i32 = channel;
-    MAP[idx as usize]
+fn cubeb_channel_to_pa_channel(channel: ffi::cubeb_channel) -> pa_channel_position_t {
+    use cubeb_backend::ffi::*;
+    assert_ne!(channel, CHANNEL_INVALID);
+
+    match channel {
+        // CHANNEL_MONO => PA_CHANNEL_POSITION_MONO,
```

Comment:
Why's this commented out?

Reply:
Because it is handled by the `_ => PA_CHANNEL_POSITION_MONO` case.  (clippy complains that two patterns match to return the same value)